addons/stock/stock.py

```diff
@@ -2145,6 +2145,9 @@ def attribute_price(self, cr, uid, move, context=None):
         """
         if not move.price_unit:
             price = move.product_id.standard_price
+            from_unit = move.product_tmpl_id.uom_id.id
+            to_unit = move.product_uom.id
+            price = self.pool['product.uom']._compute_price(cr, uid, from_unit, price, to_uom_id=to_unit)
```

Comment:
We cannot do that.  The calculation and update of the average cost price depends on this and it should be in the product UoM.  It will also change the default behaviour too much.